addons/binding/org.openhab.binding.helios/src/main/java/org/openhab/binding/helios/handler/HeliosHandler213.java

```diff
@@ -0,0 +1,962 @@
+                        ThingBuilder thingBuilder = editThing();
+                        ChannelTypeUID enablerUID = new ChannelTypeUID(BINDING_ID, SWITCH_ENABLER);
+                        ChannelTypeUID triggerUID = new ChannelTypeUID(BINDING_ID, SWITCH_TRIGGER);
+
+                        Channel channel = ChannelBuilder
+                                .create(new ChannelUID(getThing().getUID(), "switch" + aSwitch.id + "active"), "Switch")
+                                .withType(enablerUID).build();
+                        thingBuilder.withChannel(channel);
+                        channel = ChannelBuilder
+                                .create(new ChannelUID(getThing().getUID(), "switch" + aSwitch.id), "Switch")
+                                .withType(triggerUID).build();
+                        thingBuilder.withChannel(channel);
+                        updateThing(thingBuilder.build());
+
+                    }
+                }
+            }
+
+            List<RESTPort> ports = getPorts();
+
+            if (ports != null) {
+                for (RESTPort aPort : ports) {
+                    logger.debug("Adding a channel to the Helios IP Vario '{}' for the IO port with id '{}'",
+                            getThing().getUID().toString(), aPort.port);
+                    ThingBuilder thingBuilder = editThing();
+                    ChannelTypeUID triggerUID = new ChannelTypeUID(BINDING_ID, IO_TRIGGER);
+
+                    Map<String, String> channelProperties = new HashMap<String, String>();
+                    channelProperties.put("type", aPort.type);
+
+                    Channel channel = ChannelBuilder
+                            .create(new ChannelUID(getThing().getUID(), "io" + aPort.port), "Switch")
+                            .withType(triggerUID).withProperties(channelProperties).build();
+                    thingBuilder.withChannel(channel);
+                    updateThing(thingBuilder.build());
+                }
+            }
+
+        }
+
+    };
+
+    protected Runnable resetRunnable = new Runnable() {
+
+        @Override
+        public void run() {
+            logger.debug("Resetting the Helios IP Vario handler for '{}'", getThing().getUID());
+            dispose();
+            initialize();
+        }
+
+    };
+
+    protected Runnable logRunnable = new Runnable() {
+
+        @Override
+        public void run() {
+            if (getThing().getStatus() == ThingStatus.ONLINE) {
+
+                if (logSubscriptionID == 0) {
+                    logSubscriptionID = subscribe();
+                }
+
+                if (logSubscriptionID != 0) {
+                    List<RESTEvent> events = pullLog(logSubscriptionID);
+
+                    for (RESTEvent event : events) {
+
+                        Date date = new Date();
+                        SimpleDateFormat DATE_FORMATTER = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss");
```

Comment:
I'd advise to move `SimpleDateFormat` construction outside of the loop for better performance